src/item.cpp

```diff
@@ -2875,12 +2875,34 @@ void item::final_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
 
         const bionic_id bid = type->bionic->id;
 
+        const std::vector<itype_id> &fuels = bid->fuel_opts;
+        if( !fuels.empty() ) {
+            std::string fuel_string;
+            const int &fuel_numb = fuels.size();
+            fuel_string = "<info>" + item( fuels.front() ).tname() + "</info>";
```

Comment:
```suggestion
            fuel_string = "<info>" + itype( fuels.front() ).nname() + "</info>";
```
No need to construct an actual item here.

Also, it's better to use `enumerate_as_string` to join the fuel names:

```
enumerate_as_string( fuels.begin(), fuels.end(), []( const itype_id &id ) -> std::string {
    return "<info>" + itype( id ).nname() + "</info>";
} );
```

Reply:
It seems that I can't do `itype( id)`

Reply:
Oops, right. I think it should be `item_controller->find_template( id )->nname()` then. (find_template doesn't return null, so it can be directly dereferenced).